engine/execution/state/delta/storage_snapshot.go

```diff
@@ -0,0 +1,64 @@
+package delta
+
+import (
+	"github.com/onflow/flow-go/model/flow"
+)
+
+type StorageSnapshot interface {
+	Get(id flow.RegisterID) (flow.RegisterValue, error)
+}
+
+type EmptyStorageSnapshot struct{}
+
+func (EmptyStorageSnapshot) Get(
+	id flow.RegisterID,
+) (
+	flow.RegisterValue,
+	error,
+) {
+	return nil, nil
+}
+
+type ReadFuncStorageSnapshot struct {
+	ReadFunc func(flow.RegisterID) (flow.RegisterValue, error)
+}
+
+func NewReadFuncStorageSnapshot(
+	readFunc func(flow.RegisterID) (flow.RegisterValue, error),
+) StorageSnapshot {
+	return &ReadFuncStorageSnapshot{
+		ReadFunc: readFunc,
+	}
+}
+
+func (storage ReadFuncStorageSnapshot) Get(
+	id flow.RegisterID,
+) (
+	flow.RegisterValue,
+	error,
+) {
+	return storage.ReadFunc(id)
+}
+
+type Peeker interface {
+	Peek(id flow.RegisterID) (flow.RegisterValue, error)
+}
+
+type PeekerStorageSnapshot struct {
+	Peeker
+}
+
+func NewPeekerStorageSnapshot(peeker Peeker) StorageSnapshot {
```

Comment:
isn't this just `NewReadFuncStorageSnapshot(peeker.Peek)`? This implementation is not really necessary IMO.

Reply:
good point.  i'll update this